### ISSUE TITLE
Docker daemon allocates tty by default in Ubuntu 14.04

### ISSUE THREAD

**Issue by mausch** [moby/moby#17442]
On a fresh Ubuntu 14.04 install, the docker daemon logs to `/var/log/upstart/docker.log` in TTY format e.g.:

```
ESC[34mINFOESC[0m[0000] Listening for HTTP on unix (/var/run/docker.sock) 
ESC[34mINFOESC[0m[0000] Option DefaultDriver: bridge                 
ESC[34mINFOESC[0m[0000] Option DefaultNetwork: bridge                
/var/run/docker.sock is up
ESC[34mINFOESC[0m[0000] Firewalld running: false
```

I don't think this [should be the default](https://github.com/docker/docker/issues/936#issuecomment-150619468). Without timestamps the log file is nearly useless, so I propose to change this current default behaviour. No idea what to change in the daemon config to prevent this.

`docker version`: 

```
Client:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   f4bf5c7
 Built:        Mon Oct 12 05:37:18 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   f4bf5c7
 Built:        Mon Oct 12 05:37:18 UTC 2015
 OS/Arch:      linux/amd64
```

`docker info`:

```
Containers: 1
Images: 2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 4
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-31-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 1
Total Memory: 1.225 GiB
Name: mauricio-VirtualBox
ID: K4AK:UTBS:LLHY:3LP5:4TLM:EOST:UGZ2:3QLV:KSE3:YPBB:VQQW:ZM4R
WARNING: No swap limit support
```

`uname -a`:

```
Linux mauricio-VirtualBox 3.19.0-31-generic #36~14.04.1-Ubuntu SMP Thu Oct 8 10:21:08 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Environment details: VirtualBox 4.3.20

Steps to reproduce:
1. Install Ubuntu 14.04 on a VM
2. Install Docker with `wget -qO- https://get.docker.com/ | sh` (as described in https://docs.docker.com/linux/step_one/)
3. Run `sudo usermod -aG docker $your_user`
4. Reboot
5. Run `docker run hello-world`
6. Check the log at `/var/log/upstart/docker.log`

Actual results: logs in TTY format

Expected results: logs in JSON format with timestamps

**Comment by mausch** [comment]
Sorry about that, just updated the report with the required information.

**Comment by mudverma** [comment]
Hello, I am too in need of timestamps in docker logs. Is there any update on this issue? Any workaround?

**Comment by mudverma** [comment]
@mausch  I have worked out this solution! Dirty, but solves my purpose. 
/etc/init/docker.conf:42  add this option in bold
exec "$DOCKER" -d $DOCKER_OPTS *>> /root/results/docker_main.log 2>&1*

Sample output:

time="2016-01-17T05:31:55.702544849-06:00" level=debug msg="Starting container 8cdfd2d9f2daf9ee41d43878daa494ba45ecbb5504e87e8bbf74d87fe95e66dd" 
time="2016-01-17T05:31:55.849906122-06:00" level=info msg="Daemon has completed initialization" 
time="2016-01-17T05:31:55.849974818-06:00" level=info msg="Docker daemon" commit=696147b execdriver=native-0.2 graphdriver=aufs version=1.8.0-rc3

**Comment by thaJeztah** [comment]
@mudverma I think this will be fixed by https://github.com/docker/docker/pull/18621

**Comment by crosbymichael** [comment]
Has been resolved by a PR carrying the changes from #18621
